larray/util/misc.py

```diff
@@ -732,3 +734,23 @@ def common_type(arrays):
         return np.dtype(('U' if need_unicode else 'S', max_size))
     else:
         return object
+
+
+class LHDFStore(object):
```

Comment:
Why do we need our own HDFStore instead of using the one from pandas directly? To avoid closing the store if we were given an open store?

Reply:
Because ``HDFStore.__init__`` only accept string as path argument. Both [pandas.pytables.to_hdf](https://github.com/pandas-dev/pandas/blob/v0.22.0/pandas/io/pytables.py#L268) and [pandas.pytables.read_hdf](https://github.com/pandas-dev/pandas/blob/v0.22.0/pandas/io/pytables.py#L286) functions contains an if-else to test if argument ``path_or_buf`` is already an HDFStore instance or not. 

I chose to implement my own context manager ( which is actually based on the beginning of the function [pandas.pytables.read_hdf](https://github.com/pandas-dev/pandas/blob/v0.22.0/pandas/io/pytables.py#L327) )